File: test/fakers/checkout/order_item_faker.dart
import 'package:faker/faker.dart';
import 'package:sertton/core/checkout/dtos/order_item_dto.dart';

class OrderItemFaker {
  static final _faker = Faker();

  static OrderItemDto fakeDto({
    String? id,
    int? quantity,
    double? price,
    String? skuName,
    String? skuCode,
    double? skuSalePrice,
    double? skuDiscountPrice,
  }) {
    return OrderItemDto(
      id: id ?? _faker.guid.guid(),
      quantity: quantity ?? _faker.randomGenerator.integer(5, min: 1),
      price: price ?? _faker.randomGenerator.decimal(),
      skuName: skuName ?? _faker.food.dish(),
      skuCode: skuCode ?? _faker.guid.guid(),
      skuSalePrice: skuSalePrice ?? _faker.randomGenerator.decimal(),
      skuDiscountPrice: skuDiscountPrice ?? _faker.randomGenerator.decimal(),
    );
  }

  static List<OrderItemDto> fakeManyDto({int count = 3}) {
    return List.generate(count, (_) => fakeDto());
  }
}

File: src/core/checkout/entities/fakers/order-item-faker.ts
import { faker } from "@faker-js/faker"

import type { OrderItem } from "../order-item"

export class OrderItemFaker {
  static fake(orderItem?: Partial<OrderItem>): OrderItem {
    const salePrice = faker.number.float({ fractionDigits: 2, min: 10, max: 1000 })

    return {
      id: faker.string.uuid(),
      price: salePrice,
      quantity: faker.number.int({ min: 1, max: 5 }),
      skuCode: faker.string.alphanumeric(8).toUpperCase(),
      skuDiscountPrice: salePrice * 0.9,
      skuName: faker.commerce.productName(),
      skuSalePrice: salePrice,
      ...orderItem,
    }
  }

  static fakeMany(count = 10, orderItem?: Partial<OrderItem>): OrderItem[] {
    return Array.from({ length: count }, () => OrderItemFaker.fake(orderItem))
  }
}
